src/OmniSharp.Host/Services/AssemblyLoader.cs

```diff
@@ -77,5 +81,31 @@ public Assembly LoadFrom(string assemblyPath)
             _logger.LogTrace($"Assembly loaded from path: {assemblyPath}");
             return assembly;
         }
+
+        public Assembly LoadFrom(string assemblyPath, bool lockAssembly)
+        {
+            if (lockAssembly) return LoadFrom(assemblyPath);
+            if (string.IsNullOrWhiteSpace(assemblyPath)) return null;
+
+            if (!AssemblyCache.TryGetValue(assemblyPath, out var assembly))
+            {
+                try
+                {
+                    var bytes = File.ReadAllBytes(assemblyPath);
+                    assembly = Assembly.Load(bytes);
+                }
+                catch (Exception ex)
+                {
+                    _logger.LogError(ex, $"Failed to load assembly from path: {assemblyPath}");
+                }
+                if (assembly != null)
```

Comment:
tiny nit: I'd put a line break before to visually separate the if statement from the catch handler.